common/agents/event_logger.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+import asyncio
+from typing import Optional
+
+import fire
+from llama_stack import LlamaStack
+from llama_stack.types import SamplingParams, ToolResponseMessage, UserMessage
+from llama_stack.types.agent_create_params import (
+    AgentConfig,
+    AgentConfigToolCodeInterpreterToolDefinition,
+    AgentConfigToolFunctionCallToolDefinition,
+    AgentConfigToolSearchToolDefinition,
+    AgentConfigToolWolframAlphaToolDefinition,
+)
+from llama_stack.types.agents import AgentsTurnStreamChunk
+from llama_stack.types.tool_param_definition_param import ToolParamDefinitionParam
+from termcolor import cprint
+
+
+class LogEvent:
+    def __init__(
+        self,
+        role: Optional[str] = None,
+        content: str = "",
+        end: str = "\n",
+        color="white",
+    ):
+        self.role = role
+        self.content = content
+        self.color = color
+        self.end = "\n" if end is None else end
+
+    def __str__(self):
+        if self.role is not None:
+            return f"{self.role}> {self.content}"
+        else:
+            return f"{self.content}"
+
+    def print(self, flush=True):
+        cprint(f"{str(self)}", color=self.color, end=self.end, flush=flush)
+
+
+class EventLogger:
```

Comment:
Maybe we can move client-side logging into the client lib instead of defining here.

Reply:
Hmm we can [patch with custom code](https://app.stainlessapi.com/docs/guides/patch-custom-code) in client SDK, however, it seems not encouraged by Stainless. Was thinking maybe we have a separate wrapper library to add these custom logging/common utilities wrapper defined inside `common`.

Reply:
Moved to `from llama_stack.lib.agents.event_logger import EventLogger`